addons/event_crm/models/event_crm.py

```diff
@@ -0,0 +1,145 @@
+            else:
+                list_type = []
+                for t in record.event_type_ids:
+                    list_type.append(t.id) if type(t.id) is type(int()) else list_type.append(t.id.origin)
+                registration_ids = self.env['event.registration'].search(['|',('event_id','=',record.event_id.id),('event_id.event_type_id','in',list_type)]).search((safe_eval(record.domain)))
+
+            record.registration_ids = registration_ids
+
+    @api.onchange('user_id')
+    def _onchange_user_id(self):
+        if self.user_id and self.user_id.sale_team_id:
+            self.sales_team_id = self.user_id.sale_team_id
+        else:
+            self.sales_team_id = False
+    
+    @api.model
+    def _check_rules(self, event_id, registration_ids):
+        """Check if the rule applied for the registration_ids created
+        and send the parameters to create leads."""
+        if len(self.sudo().registration_ids) == 0 or not self.active:
+            return
+        if event_id.company_id == self.company_id or not self.company_id:
+            if (
+                (not self.event_type_ids and not self.event_id) or 
+                (event_id.event_type_id in self.event_type_ids and not self) or 
+                (not self.event_type_ids and self.event_id == event_id) or 
+                (event_id.event_type_id in self.event_type_ids or self.event_id == event_id)
+                ):
+                values = self._prepare_create_leads_values()
+                self.sudo()._create_leads(registration_ids, values)
+
+    def _prepare_create_leads_values(self):
+        self.ensure_one()
+        values = {
+            'user_id': self.user_id.id,
+            'type': self.lead_type,
+            'team_id': self.sales_team_id.id,
+            'tag_ids': self.tag_ids,
+        }
+        return values
+
+    def _get_additional_values(self, values, registration):
+        values.update({
+            'email_from': registration.email,
+            'phone': registration.phone,
+            'mobile': registration.mobile,
+            'event_id': registration.event_id.id,
+            'referred': 'event',
+        })
+        return values
+
+    def _additional_content_description(self, registration, description):
+        return description
+
+    def _get_description_lead(self, registration_ids):
+        description = _("Other Participants:\n")
+        first_registration = False
+        for registration in self.registration_ids & registration_ids:
+            if not first_registration:
+                first_registration = registration_ids[0]
+            description += "\t%s %s %s\n" % (registration.name, registration.email, registration.phone)
+            description = self._additional_content_description(registration, description)
+        return description, first_registration
+
+    def _create_leads(self, registration_ids, vals):
+        self.ensure_one()
```

Comment:
done

Reply:
with the ensure_one, it won't be possible to make only one create for each rule.
You don't need to specify the event and the registration_ids as they are on self.
(e.g. event_id is self.event_id)
It could be cool to make something like this:

```
def _create_leads(self):
    vals_list = []
    for rule in self:
        event_id = rule.event_id
        registrations = rule.registration_ids (a priori, it's not necessary to make the intersection (self.registration_ids & registration_ids) as self.registration_ids is computed based on the rule configuration and should already be correct)
        if self.lead_creation_type == 'one_per_attendee':
            for registration in registrations:
                  ...
                 vals_list.add(lead_values)
                 ...
       else:
             ...
             vals_list.add(lead_values)
              ...
    lead = self.env['crm.lead'].create(vals_list)
```
Note that you don't need to write the lead_id on the registration as you can set the registration_id on lead_values. (many2many will link in both ways)

Reply:
done, I still need to get the registrations since it's possible that the module can be install later. So if there was registrations created before the installation of a module or a rule creation, it will create lead for this registration or mix them with the actual registrations.